policy/release/tasks.rego

```diff
@@ -2,16 +2,26 @@
 # METADATA
 # title: Tasks
 # description: >-
-#   Enterprise Contract expects that a set of tasks were included
-#   in the pipeline build for each image to be released.
-#   This package includes a set of rules to verify that the expected
-#   tasks ran in the pipeline when the image was built.
-#   Required tasks are listed by the names given to them within the Tekton
-#   Bundle image. Optionally invocation parameter of a Task can be also
-#   mandated by including the name and the value in square brackets following
-#   the name of the task. For example: ``name[PARAM=val]``. Only single parameter
-#   is supported, to assert multiple parameters repeat the required task
-#   definition for each parameter seperately.
+#   Enterprise Contract expects that a set of tasks were
+#   included in the pipeline build for each image to be
+#   released. This package includes a set of rules to verify
+#   that the expected tasks ran in the pipeline when the image
+#   was built.
+#   Required tasks for a pipeline are specified in a configuration
+#   file provided at runtime. This file features two primary
+#   sections: pipeline-required-tasks and required-tasks.
+#   The pipeline-required-tasks section lists tasks specific
+#   to particular pipelines, while required-tasks details a
+#   default or baseline set of tasks. If your pipeline corresponds
```

Comment:
> pipeline corresponds to an entry under pipeline-required-tasks

Can you elaborate on what this means?

Reply:
I tried to elaborate more just before that to give more context. Let me know if that helps. https://github.com/enterprise-contract/ec-policies/pull/977/commits/c70865d4371d1ed3a3b53741cf6cba3b58bdbb17

Reply:
Would this be better fixed by moving the `required-tasks` list to `pipeline-required-tasks.default` ?

Reply:
We could deprecate and keep the original required-tasks lists forever, but it would make the explanation much easier, right?

Reply:
I like the idea. The only downside is what if a pipeline name is `default`? We'd have to protect that somehow.

Reply:
:shrug: `__default__` then?

Reply:
Renaming the rule data key is outside the scope of EC-71. But is an important point. Their current names don't make sense. I also think that we may have a better way to do this now. For example, it is now possible to specify an [or-list](https://github.com/release-engineering/rhtap-ec-policy/blob/c9be04198008e84c49f8575b69a0a6af2a7d78eb/data/required_tasks.yml#L20) for each required Task. This should help consolidate most of the entries under `pipeline-required-tasks`, leaving fbc as its own thing. We may take this further and simply have one list for fbc and one list of anything else like we do for deprecated labels (see [fbc](https://github.com/release-engineering/rhtap-ec-policy/blob/c9be04198008e84c49f8575b69a0a6af2a7d78eb/data/rule_data.yml#L105) vs [everything else](https://github.com/release-engineering/rhtap-ec-policy/blob/c9be04198008e84c49f8575b69a0a6af2a7d78eb/data/rule_data.yml#L49)).

Reply:
The updated docs look great. Thank you!